### PR TITLE
Remove libcmt, use native ObjectiveC FFI

### PR DESCRIPTION
**Pull request by maleadt**
Replaces libcmt, and instead uses ObjectiveC.jl to directly call Metal APIs.
Obviously a WIP, and will take a while to complete.

Depends on https://github.com/JuliaInterop/ObjectiveC.jl/pull/11
Ref https://github.com/JuliaGPU/Metal.jl/issues/104

cc @habemus-papadum

### CODE REVIEW

**Comment by maleadt** [comment]
Reached a milestone, cmt is gone!
image

**Comment by maleadt** [comment]
ObjectiveC.jl takes 3 days to register, being a new package, so I'll probably merge this with the Manifest pointing to the master branch.

**Comment by maleadt** [comment]
Thanks for nerdsniping me into doing this 🙂

File: Project.toml [hunk 1]
```diff
@@ -14,18 +12,17 @@ GPUCompiler = "61eb1bfa-7361-4325-ad38-22787b887f55"
...
 Adapt = "3"
 CEnum = "0.4"
 ExprTools = "0.1"
 GPUArrays = "8.4"
 GPUCompiler = "0.16, 0.17"
-Reexport = "1.0"
 LLVM = "4.15"
 Metal_LLVM_Tools_jll = "~0.3"
-cmt_jll = "~0.2"
+Reexport = "1.0"
```
Comment (vchuravy) [inline Project.toml:27]:
```suggestion
ObjectiveC = "0.1"
Reexport = "1.0"
```

Comment (maleadt) [reply Project.toml:27]: Haha, thanks. I'll do that locally since I need to bump the Manifest, which we've committed here.

**Pull request merged**